pypeit/images/rawimage.py

```diff
@@ -195,42 +193,35 @@ def process(self, process_steps, par, bpm=bpm, flatimages=None, bias=None,
         self.par = par
         self._bpm = bpm
 
-        # For error checking
-        steps_copy = process_steps.copy()
         # Get started
         # Standard order
         #   -- May need to allow for other order some day..
-        if 'subtract_overscan' in process_steps:
+        if par['use_biasimage']:  # Bias frame, if it exists, is *not* trimmed nor oriented
+            self.subtract_bias(bias)
+        if par['use_overscan']:
             self.subtract_overscan()
-            steps_copy.remove('subtract_overscan')
-        if 'trim' in process_steps:
+        if par['trim']:
             self.trim()
-            steps_copy.remove('trim')
-        if 'orient' in process_steps:
+        if par['orient']:
             self.orient()
-            steps_copy.remove('orient')
-        if 'subtract_bias' in process_steps: # Bias frame, if it exists, is trimmed and oriented
-            self.subtract_bias(bias)
-            steps_copy.remove('subtract_bias')
-        if 'apply_gain' in process_steps:
+        if par['use_darkimage']:  # Dark frame, if it exists, is trimmed and oriented
+            self.subtract_dark(dark)
+        if par['apply_gain']:
             self.apply_gain()
-            steps_copy.remove('apply_gain')
 
```

Comment:
It might be useful to attach a string like process_steps to the pypeit image object that you build up as you execute each of these steps. I realize the hooks are in the parset now, but it is nevertheless nice to know what was done to the image.